Add Wave ML guide to docs

geomodular opened the pull request:
If PR is merged, `Wave ML` guide will be added to the docs. I included a description, current API overview, and function prototypes. We can also add a Wine tutorial to the `Getting Started` section if desired.

Resolves #683

lo5 commented on website/docs/wave-ml.md:
```diff
@@ -0,0 +1,200 @@
+
+## Functions
+
+<div className='api'>
```
The HTML in this section makes it difficult to maintain - it's better if this is reformatted as markdown, or generated and added to the API section.

geomodular replied:
> The HTML in this section makes it difficult to maintain

It's the autogenerated HTML from Wave docs to match the website API style. That's the only reason I used HTML, to match the style.

> it's better if this is reformatted as markdown, or generated and added to the API section.

True.

We cannot autogenerate it since it's in another repo. I believe we can create a script to clone the repo and autogenerate the docs. Is that the approach we want to try?

lo5 replied:
> I believe we can create a script to clone the repo and autogenerate the docs. Is that the approach we want to try?

Yes. Currently we generate the API docs from Python at build-time (the generated .md files are not checked into git). 
The `make` task can pull the repo (clone if missing), extract docs, and then build the main docusaurus site. This way, you can probably reuse the pdoc .md generator in the wave repo.

geomodular replied:
Will try that, thanks!

geomodular commented:
API for *Wave ML* is now autogenerated. It's done using the regular `make setup && make docs` targets inside `py/` directory. The version is specified here: https://github.com/h2oai/wave/blob/1b424e39dd31f54e83f4c3449b8fa045bfc8d6ef/py/Makefile#L1

The version needs to be specified as we don't want to publish the current master branch.

## Notes
- During the Makefile `setup` stage, `h2o-wave-ml` has to be installed into `venv/` so pdoc can work with dependencies.
- I've chosen API for **v0.3.0** as this is the current published version.

geomodular commented:
> During the Makefile setup stage, h2o-wave-ml has to be installed into venv/ so pdoc can work with dependencies.

I removed the *Wave ML* installation from the `setup` stage, as it's not a mandatory package. I install it during the `docs` stage.

> I've chosen API for v0.3.0 as this is the current published version.

`v0.4.0` is out so I updated this as well.

lo5 reviewed (approved):
Thanks @geomodular!
